typed-racket-lib/typed-racket/base-env/prims-struct.rkt

```diff
@@ -81,12 +81,22 @@
                                               (~bind [ecname #f]))
                                         (~and (~seq #:extra-constructor-name ecname)
                                               (~bind [cname #f]))))
+                        (~optional (~seq #:type-name type:id))
                         ;; FIXME: unsound, but relied on in core libraries
                         ;; #:guard ought to be supportable with some work
                         ;; #:property is harder
                         (~optional (~seq #:guard guard:expr))
                         (~seq #:property prop:expr prop-val:expr))
-                   ...)))
+                   ...)
+             #:attr untyped #`(#,@(if (attribute mutable?) #'(#:mutable) #'())
```

Comment:
The simplest way would be to use a parameterized splicing syntax class that has the keyword in a list as an attribute, if present, and an empty list otherwise. Also binding it to an attribute of `...` depth 1 would let you write the results as something like:

```
mutable?.kw ...
transparent?.kw ...
```
and so on.